### PR TITLE
Produce 2 binaries instead of one

### PR DESCRIPTION
**Pull request by Itxaka**
This generates a different binary for the register command as to not
bundle it with the full operator.

On the makefile separated builds have been done for operator and
registry.

On the docker image a different container is created for the register
and pushed to a different repo as well.

For releases gorelease generates 2 different binaries.

Signed-off-by: Itxaka <igarcia@suse.com>

### CODE REVIEW

**Comment by Itxaka** [comment]
Fixes https://github.com/rancher/elemental-operator/issues/72

File: .github/workflows/docker.yaml [hunk 1]
```diff
@@ -50,3 +58,17 @@ jobs:
...
+          tags: ${{ steps.meta-register.outputs.tags }}
+          labels: ${{ steps.meta-register.outputs.labels }}
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          target: elemental-register
+          push: true
+          build-args: |
+            TAG=${{ steps.export_tag.outputs.operator_tag }}
+            COMMITDATE=${{ steps.export_tag.outputs.commit_date }}
+            COMMIT=${{ github.sha }}
```
Comment (mjura) [inline .github/workflows/docker.yaml:74]: Missing new line character

File: Dockerfile [hunk 2]
```diff
@@ -17,6 +18,18 @@ RUN go build  \
...
+
 FROM scratch as elemental-operator
 COPY --from=build /usr/sbin/elemental-operator /usr/sbin/elemental-operator
-ENTRYPOINT ["/usr/sbin/elemental-operator"]
\ No newline at end of file
+ENTRYPOINT ["/usr/sbin/elemental-operator"]
+
+FROM scratch as elemental-register
+COPY --from=build /usr/sbin/elemental-register /usr/sbin/elemental-register
+ENTRYPOINT ["/usr/sbin/elemental-register"]
```
Comment (mjura) [inline Dockerfile:35]: Missing new line character

**Review by mjura — APPROVED** [review]
It looks good besides new line characters

**Pull request merged**